### ISSUE TITLE
Amount of quantization

### ISSUE THREAD

**Issue by danishnazir** [InterDigitalInc/CompressAI#256]
Hello,

Is it possible to somehow defined the level of quantization during training? For instance one can use the following code for hard quantization during training,
```
from .compressai.ops.ops import quantize_ste
_, y_likelihoods = self.gaussian_conditional(y, scales_hat)
y_hat = quantize_ste(y)
```
then if we look inside quantize_ste function, we have
`return (torch.round(x) - x).detach() + x`
This is basically dependant on the input` x`, however, I am not really sure what level of quanitzation will be applied to latent variable `y`.
Is this possible to know during training?

**Comment by YodaEmbedding** [comment]
The latent `y` is quantized uniformly to bins of width 1.

This is done in one of the following ways:

1. Uniform noise $\mathcal{U}(-\frac{1}{2}, \frac{1}{2})$ during training, and rounding during evaluation.
2. STE, which rounds, but passes through the gradients during the backwards pass:
    ```python
    ste(y) = round(y)   during forward
    ste(y) = y           during backward
    ```

**Comment by danishnazir** [comment]
Thanks for your response. For the case of STE, if we somehow want to increase the quantization level, what would be the solution? Can we do something like `ste(y) = round(y) * 2` and then keep the gradients same in the backward as you said?
